### ISSUE TITLE
Couldn't find source adapter server1.conn4.child1 etc etc

### ISSUE THREAD

**Issue by jasonwilliams** [firefox-devtools/vscode-firefox-debug#161]
![image](https://user-images.githubusercontent.com/936006/69240129-b865d180-0b93-11ea-8f5c-bbd16d12a989.png)

Any idea what this problem is?
My paths seem correct.

Always seems to be regenerator-runtime too

**Comment by hbenl** [comment]
That's a bug in this extension (not in your configuration, it has nothing to do with the paths). I've seen this before, but couldn't reproduce it. Do you have a test case for reproducing it?

**Comment by jasonwilliams** [comment]
@hbenl i will try to get you reproduction steps.
it could be related to this https://github.com/zeit/next.js/issues/9471 which i had happening at the same time

**Comment by hbenl** [comment]
> do you know what the "couldn't find adapter" means in this case?

That is an internal error message from my code, I've seen this before but thought I had fixed this in version 2.3.3 (at least I didn't see this anymore since then).
Could you tell me a bit more about when this happens: hitting an exception or a breakpoint or while stepping, does it happen sporadically or can you reproduce it reliably in your app etc.

**Comment by jasonwilliams** [comment]
* It happens on a breakpoint
* I can reproduce it every time on this breakpoint

**Comment by hbenl** [comment]
(sorry for the delay, I've been busy with some other fixes)
Your screenshot looks like you've been trying to step out of a function when this happened, is that correct? Does it only happen when stepping out or does it also happen before that?
Is the function that you're stepping out of an `async` function? Is it in your code (as opposed to some library code in `node_modules`)?

It would be really helpful if you could create a log file by adding
```
    "log": {
        "fileName": "${workspaceFolder}/log.txt",
        "fileLevel": {
            "default": "Debug"
        }
    }
```
(or similar) to your `launch.json` and share it here.

**Comment by vinnyjames** [comment]
I've seen the same thing.  In the attached log, the first few times the breakerpoint hits.  Each time it hits, I do these steps:

1. click to turn the breakpoint off
2. click to turn the breakpoint on
3. press ctrl-s to save (browser reloads autmatically via reloadOnChange launch setting)
4. observe it either working correctly or producing an error like "Couldn't find source adapter for server1.conn0.child1/source42"

[vscode-debug.log](https://github.com/firefox-devtools/vscode-firefox-debug/files/4189302/vscode-debug.log)

**Comment by vinnyjames** [comment]
I noticed it happens more frequently when editing the code and saving without letting the debugger go.  I tried for about a minute to reproduce while letting it go "Continue" each time to no avail.  Once I started saving without Continue it happened right away.  It also seems to actually be attached just without any visual indication.  Pressing F10 to Step Over activates the debugger on the next line right after the breakpoint.

![ff debug double size](https://user-images.githubusercontent.com/5167824/74294611-4bc22500-4cf3-11ea-824f-cfdf04b4399a.gif)

**Comment by vinnyjames** [comment]
Found an easier way to reproduce it and in a very simple single-file project:
1. add a few breakpoints
2. save
3. remove one of the breakpoints
4. save

total overkill: https://github.com/vinnyjames/vscode-firefox-debug_issues161

[161.txt](https://github.com/firefox-devtools/vscode-firefox-debug/files/4189489/161.txt)

![ff-debug](https://user-images.githubusercontent.com/5167824/74296122-d4db5b00-4cf7-11ea-91f5-d81b1afbbca2.gif)

**Comment by hbenl** [comment]
Sorry, I forgot to mention that this is fixed in version 2.7.0.
@vinnyjames Thanks a lot for the screencast, it showed to me that I needed to reload the page in Firefox while the debugger is still paused (it never happened to me before because I always reloaded the page in Firefox after resuming the debugger).
